lib/util/http.js

```diff
@@ -144,28 +138,23 @@ const url = (strings, ...parts) => {
 
 // Checks Etag in the request if it matches serverEtag then returns 304
 // Otherwise executes given function 'fn'
-const withEtag = (serverEtag, fn) => (request, response) => {
-
-  response.set('ETag', `"${serverEtag}"`);
+const withEtag = (serverEtag, fn, always=true) => (request, response) => {
+  if (always) response.set('ETag', `"${serverEtag}"`);
```

Comment:
I've tried to clarify by:

1. adding a comment about `always`, and
2. splitting the exported `withEtag()` function to add an additional `withEtagOnMatch()`

Does this help?  Perhaps `withEtagOrRedirect()` would be a clearer name for the new function?

Reply:
I mostly follow but now I'm confused by the `!always` line below.  I saw online that a 304 not modified request should always be sending back the etag in the header anyway. 

But also in the s3 redirect case, `always` will be false, so:
- dont set etag
- check client etag
- if it matches, DO set etag and return 304 not modified

Does s3 make its own etag? Will these ever match and would we ever return a 304 in place of a 307?

Reply:
> Does s3 make its own etag? Will these ever match and would we ever return a 304 in place of a 307?

Great question.  If a client sees an odk-central URL for an s3-backed blob, odk-central needs to ensure that the final ETag provided to the user is the same regardless of the binary data source.

This means either both systems need to generate the same ETag for the same data, or odk-central needs to substitute its ETag in place of S3's.

Currently:

* odk-central's nginx should be transparently following 307 redirects for s3-backed content
* both S3 and odk-central-backend use MD5(blob-content) as a strong ETag

This means that the _current_ implementation should work fine.  As there are no end-to-end central tests running both odk-central-backend and nginx, there's a risk this could break in future.

Reply:
> What's this `always` flag about?

I've reversed this flag, renaming it as `onlyOnMatch`, and updated the comments.  Is the intention clearer now?